File: Study-plan- Android/app/src/main/java/com/example/studyplan/TextPreprocessor.java
package com.example.studyplan;

public class TextPreprocessor {

    public static String clean(String rawInput) {
        if (rawInput == null || rawInput.trim().isEmpty()) return "";

        String result = rawInput
                .replaceAll("(?i)当然可以[，,].*?[：:]", "")
                .replaceAll("(?i)以下是.*?学习计划[：:]", "")
                .replaceAll("(?i)好的[，,].*?[：:]", "")
                .replaceAll("(?i)收到[，,].*?[：:]", "")
                .replaceAll("(?i)没问题[，,].*?[：:]", "");

        result = result.trim()
                .replace("，", ",")
                .replace("；", ";");

        return result;
    }
}

File: Study-plan-iOS/Study-plan/Helpers/TextPreprocessor.swift
import Foundation

enum TextPreprocessor {
    static func clean(_ rawInput: String) -> String {
        var result = rawInput.trimmingCharacters(in: .whitespacesAndNewlines)
        [#"(?i)当然可以[，,].*?[：:]"#, #"(?i)以下是.*?学习计划[：:]"#, #"(?i)好的[，,].*?[：:]"#, #"(?i)收到[，,].*?[：:]"#, #"(?i)没问题[，,].*?[：:]"#].forEach { result = result.replacingOccurrences(of: $0, with: "", options: .regularExpression) }
        return result.replacingOccurrences(of: "，", with: ",").replacingOccurrences(of: "；", with: ";")
    }
}
